### PR TITLE
Minor changes for PR 252

### PR DESCRIPTION
**Pull request by akeeste**
Minor changes from the review of MHKiT-Software/MHKiT-Python#252:
- minor linting
- module docstring
- update parameter validation

### CODE REVIEW

File: mhkit/qc/upcrossing.py [hunk 1]
```diff
@@ -1,3 +1,52 @@
...
+- matplotlib: Used for plotting.
+
+Notes:
+------
+- To access the WIND Toolkit hindcast data, users need to configure `h5pyd` 
+  for data access on HSDS (see the WTK_hindcast_example notebook for more details).
+
+- While some functions perform basic checks (e.g., verifying that latitude 
+  and longitude are within a predefined region), it's essential to understand 
+  the boundaries of each region and the available parameters and elevations in the dataset.
```
Comment (mbruggs) [inline mhkit/qc/upcrossing.py]: Some of these comments don't appear relevant to this module (eg references to configuring h5pyd), just checking that we want to keep them here?

**Review by mbruggs — COMMENTED** [review]
Thanks for making these changes, Adam. Just a minor comment about the docstring.

**Comment by akeeste** [comment]
@mbruggs Oops thanks for catching that. I copied the formatting from another module and forgot to remove all the extraneous items. I'll make that change when I update the location later this week.

**Comment by akeeste** [comment]
@mbruggs I removed the extraneous doc string, moved the module into utils, and updated the places that ``upcrossing`` is imported. Let me know if you have any other suggestions. 

The files changed increased drastically because I pulled in ``develop`` so that I could edit the updated utils module. Those changes should not appear in the original #252

I'm ready to merge this, confirm things look good in #252, then merge that. Thanks!
